standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/conf/MetastoreConf.java

```diff
@@ -753,6 +755,12 @@ public enum ConfVars {
             "SQL. For some DBs like Oracle and MSSQL, there are hardcoded or perf-based limitations\n" +
             "that necessitate this. For DBs that can handle the queries, this isn't necessary and\n" +
             "may impede performance. -1 means no batching, 0 means automatic batching."),
+    DIRECT_SQL_MAX_PARAMS_IN_INSERT("metastore.direct.sql.max.parameters.in.insert",
```

Comment:
you can reuse the property: `metastore.direct.sql.batch.size`

Reply:
Have added the new parameter `metastore.direct.sql.max.parameters.in.insert` because multiple rows insert query performance would depend on the number of columns in the table too. Each table may have different column count. With this parameter, can calculate the maximum rows to insert with a single insert query for the table as show below. Number of rows in a multirow insert query would vary for each table depending upon the number of columns in the table. 

   `int maxRowsInBatch = maxParamsCount / columnCount; => max rows in the query
    int maxBatches = rowCount / maxRowsInBatch;`